substrate/runtime-support/src/storage.rs

```diff
@@ -24,8 +24,9 @@ use codec::{Slicable, KeyedVec};
 
 /// Return the value of the item in storage under `key`, or `None` if there is no explicit entry.
 pub fn get<T: Slicable + Sized>(key: &[u8]) -> Option<T> {
-	let raw = runtime_io::storage(&twox_128(key)[..]);
-	Slicable::decode(&mut &raw[..])
+	let maybe_raw = runtime_io::storage(&twox_128(key)[..]);
+	maybe_raw.map(|raw| Slicable::decode(&mut &raw[..])
+		.expect("storage should contain a decodable value if there is some entry"))
```

Comment:
storage might contain a decodable value but not necessarily for the type which was passed. there is no type-safe mapping of storage keys to types.

also seems like the zero-copy stuff from this commit has been lost: https://github.com/paritytech/polkadot/commit/064c01e8f384e267fcbafe371730c57339469fdb#diff-3baabe30cd64ea0392981bbcd238b6e5R25

Reply:
decoding for the wrong type is invalid and therefore panickable (within the runtime ofc).